src/t_stream.c

```diff
@@ -744,8 +744,14 @@ int64_t streamTrim(stream *s, streamAddTrimArgs *args) {
 
         /* Check if we can remove the whole node. */
         int remove_node;
-        streamID master_id = {0}; /* For MINID */
-        if (trim_strategy == TRIM_STRATEGY_MAXLEN) {
+        streamID master_id = {0};
+        /* Read the master ID from the radix tree key. */
+        streamDecodeID(ri.key, &master_id);
+        /* Read last ID. */
+        streamID last_id;
+        lpGetEdgeStreamID(lp, 0, &master_id, &last_id);
```

Comment:
these lines are now overlapping with what's in the `else` below, so maybe they should just be moved instead?

Reply:
removed the statements from else part.